mawabot/cogs/messages.py

```diff
@@ -0,0 +1,85 @@
+#
+# cogs/messages.py
+#
+# mawabot - Maware's selfbot
+# Copyright (c) 2017 Ma-wa-re, Ammon Smith
+#
+# mawabot is available free of charge under the terms of the MIT
+# License. You are free to redistribute and/or modify it under those
+# terms. It is distributed in the hopes that it will be useful, but
+# WITHOUT ANY WARRANTY. See the LICENSE file for more details.
+#
+
+''' Has several commands that deal with messages and text '''
+
+import discord
+from discord.ext import commands
+
+__all__ = [
+    'setup',
+]
+
+class Messages:
+    __slots__ = (
+        'bot',
+    )
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    # Helper methods
+    @staticmethod
+    async def _get_messages(channel, ids):
+        ''' Gets a list of the messages with the given IDs, in that order '''
+
+        messages = []
+        async for msg in channel.history():
+            try:
+                messages.append((ids.index(msg.id), msg))
+            except ValueError:
+                pass
+
+        return map(lambda t: t[1], sorted(messages))
+
+    # Commands
+    @commands.command()
+    async def hit(self, ctx, *, content: str):
+        ''' Sends the given message and then immediately deletes it '''
+
+        fut = ctx.message.delete()
+        await ctx.send(content=content, delete_after=0)
+        await fut
+
+    @commands.command()
+    async def quote(self, ctx, *ids: int):
+        ''' Quotes the given post(s) '''
+
+        fut = ctx.message.delete()
+        to_quote = await self._get_messages(ctx.channel, ids)
+        for msg in to_quote:
+            embed = discord.Embed(type='rich', description=msg.content)
+            embed.set_author(name=msg.author.display_name)
+            await ctx.send(embed=embed)
+        await fut
+
+    @commands.command()
+    async def dump(self, ctx, *ids: int):
+        ''' Outputs the literal contents of the given post(s) '''
+
+        fut = ctx.message.delete()
```

Comment:
Instead of doing this for each command you can make a cog command after invoke hook, this is run after every command regardless of if it was successful ([Before and After Invocation Hooks](http://discordpy.readthedocs.io/en/rewrite/migrating.html#before-and-after-invocation-hooks)).